### ISSUE TITLE
Fix #10607 seems to have broken my ArrayHelper::multisort()

### ISSUE THREAD

**Issue by bluezed** [yiisoft/yii2#12966]
Since the above fix has been implemented one of my multisort-calls is not sorting correctly any more. If I comment out the fix it works again.

### What steps will reproduce the problem?
I have an array of user-objects and roles. Basically looking like this only with 100s of entries in no particular order:
```
$userMap =
array (
  5143 => 
  array (
    'user' => 
    app\models\LdapUser::__set_state(array(
       'user_settings' => 
      array (
      ),
       '_isAdmin' => NULL,
       '_attributes' => 
      array (
        'id' => 5143,
        'uid' => 'martin',
        'full_dn' => 'uid=martin,ou=People,dc=ldap',
        'full_name' => 'Martin Bower',
        'last_name' => 'Bower',
        'first_name' => 'Martin',
        'email' => 'martin.bower@example.com',
        'phone' => NULL,
        'room' => NULL,
        'company' => NULL,
        'ssh_key' => NULL,
        'is_person' => 1,
        'is_active' => 1,
        'auth_key' => '',
        'settings' => '[]',
      ),
       '_oldAttributes' => 
      array (
        'id' => 5143,
        'uid' => 'martin',
        'full_dn' => 'uid=martin,ou=People,dc=ldap',
        'full_name' => 'Martin Bower',
        'last_name' => 'Bower',
        'first_name' => 'Martin',
        'email' => 'martin.bower@example.com',
        'phone' => NULL,
        'room' => NULL,
        'company' => NULL,
        'ssh_key' => NULL,
        'is_person' => 1,
        'is_active' => 1,
        'auth_key' => '',
        'settings' => '[]',
      ),
       '_related' => 
      array (
      ),
       '_errors' => NULL,
       '_validators' => NULL,
       '_scenario' => 'default',
       '_events' => 
      array (
      ),
       '_behaviors' => 
      array (
      ),
    )),
    'roles' => 
    array (
      0 => 'General Users',
    ),
  ),
  4005 => 
  array (
    'user' => 
    app\models\LdapUser::__set_state(array(
       'user_settings' => 
      array (
      ),
       '_isAdmin' => NULL,
       '_attributes' => 
      array (
        'id' => 4005,
        'uid' => 'fred',
        'full_dn' => 'uid=fred,ou=People,dc=ldap',
        'full_name' => 'Fred Github',
        'last_name' => 'Github',
        'first_name' => 'Fred',
        'email' => 'fred.github@example.com',
        'phone' => NULL,
        'room' => NULL,
        'company' => NULL,
        'ssh_key' => NULL,
        'is_person' => 1,
        'is_active' => 1,
        'auth_key' => '',
        'settings' => '[]',
      ),
       '_oldAttributes' => 
      array (
        'id' => 4005,
        'uid' => 'fred',
        'full_dn' => 'uid=fred,ou=People,dc=ldap',
        'full_name' => 'Fred Github',
        'last_name' => 'Github',
        'first_name' => 'Fred',
        'email' => 'fred.github@example.com',
        'phone' => NULL,
        'room' => NULL,
        'company' => NULL,
        'ssh_key' => NULL,
        'is_person' => 1,
        'is_active' => 1,
        'auth_key' => '',
        'settings' => '[]',
      ),
       '_related' => 
      array (
      ),
       '_errors' => NULL,
       '_validators' => NULL,
       '_scenario' => 'default',
       '_events' => 
      array (
      ),
       '_behaviors' => 
      array (
      ),
    )),
    'roles' => 
    array (
      0 => 'General Users',
    ),
  ),
 [ ... ]
```

With this array I call 
`ArrayHelper::multisort($userMap, ['last_name', 'first_name'], SORT_ASC, SORT_STRING);`
because I want the array sorted by users last_name and first_name.

### What is the expected result?
The entries in the array should be sorted by the users' last and first names.

### What do you get instead?
I get a somewhat sorted array but it's not consistent and a lot of entries are totally in the wrong place.

### Additional info

If I comment out the above fix the sorting is correct again.

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10
| PHP version      | 5.5.14
| Operating system | SLES12

**Comment by samdark** [comment]
#10607

**Comment by bluezed** [comment]
I've done a bit more investigation... Looks like it was my error rather than anything wrong with the multisort.

Turns out even with the fix commented out the sort is still incorrect but not immediately obvious. The fix just made it more obvious ;) 

So after reading the source code for multisort, getColumns and getValue I found my error. 
I was assuming that the $key array would be evaluated and used to look inside the objects in the array and I never questioned that because it appeared to work fine. But in reality I needed to specify the actual class attribute in the $key array. So changing the call to this
`ArrayHelper::multisort($userMap, ['user.last_name', 'user.first_name'], SORT_ASC, SORT_STRING);`
solves the issue and it works fine with and without the fix.

So sorry for wasting your time with this.
